nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/GrokParser.java

```diff
@@ -0,0 +1,243 @@
+        descriptors.add(GROK_EXPRESSION);
+        descriptors.add(GROK_PATTERN_FILE);
+        descriptors.add(DESTINATION);
+        descriptors.add(CHARACTER_SET);
+        descriptors.add(MAX_BUFFER_SIZE);
+        this.descriptors = Collections.unmodifiableList(descriptors);
+
+        final Set<Relationship> relationships = new HashSet<Relationship>();
+        relationships.add(REL_MATCH);
+        relationships.add(REL_NO_MATCH);
+        this.relationships = Collections.unmodifiableSet(relationships);
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return this.relationships;
+    }
+
+    @Override
+    public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return descriptors;
+    }
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+
+        final int maxBufferSize = context.getProperty(MAX_BUFFER_SIZE).asDataSize(DataUnit.B).intValue();
+        buffer = new byte[maxBufferSize];
+
+        try{
+            grok = Grok.create(context.getProperty(GROK_PATTERN_FILE).getValue());
+            grok.compile(context.getProperty(GROK_EXPRESSION).getValue());
+        }catch (GrokException e){
+            getLogger().error("Failed to initialize ExtractGrok due to: ", e);
+        }
+
+
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        FlowFile flowFile = session.get();
+        if ( flowFile == null ) {
+            return;
+        }
+
+        final Charset charset = Charset.forName(context.getProperty(CHARACTER_SET).getValue());
+        final Map<String, String> grokResults = new HashMap<>();
+        final byte[] byteBuffer = buffer;
+        session.read(flowFile, new InputStreamCallback() {
+            @Override
+            public void process(InputStream in) throws IOException {
+                StreamUtils.fillBuffer(in, byteBuffer, false);
+            }
+        });
+        final long len = Math.min(byteBuffer.length, flowFile.getSize());
+        final String contentString = new String(byteBuffer, 0, (int) len, charset);
+
+
+        final Match gm = grok.match(contentString);
+        gm.captures();
+        for(Map.Entry<String,Object> entry: gm.toMap().entrySet()){
+            if(null != entry.getValue() ) {
+                grokResults.put(entry.getKey(), entry.getValue().toString());
+            }
+        }
+
+        if (grokResults.isEmpty()) {
+            session.transfer(flowFile, REL_NO_MATCH);
+            getLogger().info("Did not match any Grok Expressions for FlowFile {}", new Object[]{flowFile});
+            return ;
+        }
+        switch (context.getProperty(DESTINATION).getValue()){
+            case DESTINATION_ATTRIBUTE:
+
+
+                flowFile = session.putAllAttributes(flowFile, grokResults);
+
+                session.getProvenanceReporter().modifyAttributes(flowFile);
```

Comment:
It's probably worth reporting this along with the amount of time that it took to update the attributes. This can be very nice information to have if troubleshooting a flow.